src/routes/clusters.py

```diff
@@ -0,0 +1,86 @@
+"""
+Routes for Spark cluster management.
+"""
+
+import logging
+import math
+
+from fastapi import APIRouter, Depends, status
+
+from src.service import kb_auth
+from src.service.dependencies import auth
+from src.service.exceptions import ConfigurationLimitExceededError
+from src.service.kb_auth import AdminPermission
+from src.service.models import (
+    DEFAULT_MASTER_MEMORY,
+    DEFAULT_WORKER_MEMORY,
+    SparkClusterConfig,
+    SparkClusterCreateResponse,
+)
+from src.spark_manager import KubeSparkManager
+
+logger = logging.getLogger(__name__)
+
+# Create a router for cluster endpoints
+router = APIRouter(prefix="/clusters", tags=["clusters"])
+
+
+@router.post(
+    "",
+    response_model=SparkClusterCreateResponse,
+    status_code=status.HTTP_201_CREATED,
+    summary="Create a new Spark cluster",
+    description="Creates a new Spark cluster for the authenticated user with the specified configuration.",
+)
+async def create_cluster(
+    config: SparkClusterConfig, user: kb_auth.KBaseUser = Depends(auth)
+) -> SparkClusterCreateResponse:
+    """Create a new Spark cluster for the authenticated user."""
+
+    manager = KubeSparkManager(
+        username=str(user.user),
+    )
+
+    # Validate config against defaults for non-admin users
+    if user.admin_perm != AdminPermission.FULL:
+        default_config = SparkClusterConfig(
+            worker_memory=DEFAULT_WORKER_MEMORY,  # type: ignore[assignment]
+            master_memory=DEFAULT_MASTER_MEMORY,  # type: ignore[assignment]
+        )
```

Comment:
I don't understand why this is setting the worker and master memory - aren't those the default values, so this is a noop?

Reply:
yea. I thought the same. But if I don't explicitly do that the below comparison (`config.worker_memory > default_config.worker_memory`) raise an error.

```
    |   File "/app/src/routes/clusters.py", line 54, in create_cluster
    |     or config.worker_memory > default_config.worker_memory
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    | TypeError: '>' not supported between instances of 'ByteSize' and 'str'
```

If I make worker_memory defaults to ByteSize(DEFAULT_WORKER_MEMORY), the `ge`/`le` won't work - as I mentioned it in the previous PR.